Makefile

```diff
@@ -190,6 +190,13 @@ clint: build/.ran-cmake
 clint-full: build/.ran-cmake
 	+$(BUILD_CMD) -C build clint-full
 
+clang-tidy: generate-compilation-db
+	bash scripts/run_clang_tidy.sh $(FILE)
+
+generate-compilation-db: build/.ran-cmake
+	ninja -C build -t compdb C_COMPILER__nvim_$(CMAKE_BUILD_TYPE) > build/compile_commands.json	
+	ln -sf build/compile_commands.json .
```

Comment:
looks useful 👀

Reply:
Yeah, it's the same one I mentioned in #10632

This is essential if we want to run this in the CI and not have it analyze third-party and other non-essential files. 
Let me know if you feel that it's missing some files. You can verify the list quickly with `jq`.

```bash
# you can change the filter or drop it all together
cat compile_commands.json | jq -c '.[] | select( .file | contains("src"))' | wc -l
```

Reply:
Is there a particular reason for creating a symbolic link to `compile_commands.json` in the base directory?
Clangd for example is already configured to search for it in `build/` through `.clangd`.

Reply:
My bad, I didn't notice it. 

It's just the standard way to let the other tools find it without hard-coding the build path. I wanted to use $(BUILD_DIR) but it's not defined.

```make
ln -sf $(BUILD_DIR)/compile_commands.json .
```

There are a few ways for cmake to use it:
1) your typical 
```shell
cd $BUILD_DIR && cmake .. 
```
2) by setting the `CMAKE_BINARY_DIR=..` variable
2) my favorite, but I believe it's relatively new so I need to check which is the oldest supported cmake version
```shell
cmake -B $BUILD_DIR
```